### PR TITLE
chore: some CMake cleanup

### PR DESCRIPTION
**Pull request by henryiii**
Some things I noticed when trying something locally.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +5,6 @@
 # The full license is in the file LICENSE, distributed with this software.
 
 cmake_minimum_required(VERSION 3.16)
-cmake_policy(SET CMP0025 NEW) # Introduced in cmake 3.0
-cmake_policy(SET CMP0077 NEW) # Introduced in cmake 3.13
```
Comment (henryiii) [inline CMakeLists.txt:9]: Setting the minimum version of CMake sets all policies to NEW that were introduced on or before that value. (Also true if you set a version range, like 3.16...3.30)

File: docs/source/developer_zone/dev_environment.rst [hunk 2]
```diff
@@ -37,7 +37,7 @@ Install development dependencies
 .. code:: bash
 
     micromamba create -n mamba -c conda-forge -f dev/environment-dev.yml
-    micromamba activate -n mamba
+    micromamba activate mamba
```
Comment (henryiii) [inline docs/source/developer_zone/dev_environment.rst:40]: There's no `-n` here.

File: libmamba/CMakeLists.txt [hunk 3]
```diff
@@ -405,15 +405,15 @@ set(
...
-find_package(fmt REQUIRED)
-find_package(spdlog REQUIRED)
-find_package(tl-expected REQUIRED)
-find_package(nlohmann_json REQUIRED)
-find_package(simdjson REQUIRED)
-find_package(yaml-cpp REQUIRED)
-find_package(reproc REQUIRED)
-find_package(reproc++ REQUIRED)
-find_package(Libsolv REQUIRED)
+find_package(fmt CONFIG REQUIRED)
```
Comment (henryiii) [inline libmamba/CMakeLists.txt:408]: Using `CONFIG` explicitly makes the error message better. I didn't go through every one to figure out whether they were CONFIG or MODULE, but these are some of the first ones hit, so error message is important for these.

File: libmambapy/CMakeLists.txt [hunk 4]
```diff
@@ -7,20 +7,16 @@ cmake_minimum_required(VERSION 3.18.2)
...
 if(NOT TARGET mamba::libmamba)
-    find_package(libmamba REQUIRED)
+    find_package(libmamba CONFIG REQUIRED)
     set(libmamba_target mamba::libmamba-dyn)
 else()
     set(libmamba_target mamba::libmamba)
 endif()
 
-find_package(Python COMPONENTS Interpreter Development)
+find_package(Python COMPONENTS Interpreter Development.Module)
```
Comment (henryiii) [inline libmambapy/CMakeLists.txt:19]: The default `Development` looks for the `Embed` target too, which isn't required and sometimes (like on manylinux) not available.

**Review by jjerphan — APPROVED** [review]
Hi @henryiii,

Thank you for bringing your expertise in build system and meta-build systems with this contribution!

Edit: I don't know whether the failures on Windows are related to that, and I can't debug it.

**Comment by henryiii** [comment]
The PR after this one has the same windows error, and the one before this one claims to fix something on Windows.

**Comment by jjerphan** [comment]
#3555 fixes an issue related to tests under Windows.

The error here is due to vcpkg which is used for the static builds of micromamba on Windows — this might be beyond our control.

**Review by JohanMabille — APPROVED** [review]

**Comment by JohanMabille** [comment]
This looks like a network issue, libxml2 is available at [https://gitlab.gnome.org//GNOME/libxml2/-/archive/v2.11.9/libxml2-v2.11.9.tar.gz](https://gitlab.gnome.org//GNOME/libxml2/-/archive/v2.11.9/libxml2-v2.11.9.tar.gz:)

File: libmamba/CMakeLists.txt [hunk 5]
```diff
@@ -539,7 +539,7 @@ macro(libmamba_create_target target_name linkage output_name)
             set(CMAKE_PREFIX_PATH "$ENV{VCPKG_ROOT}/installed/x64-windows-static-md/")
 
             # For Windows we have a vcpkg based build system right now.
-            find_package(LibArchive REQUIRED)
+            find_package(LibArchive CONFIG REQUIRED)
```
Comment (JohanMabille) [inline libmamba/CMakeLists.txt:542]: I t looks like LibArhcive does not provide a cmake config file on Windows.

Comment (JohanMabille) [inline libmamba/CMakeLists.txt:542]:
```suggestion
            find_package(LibArchive REQUIRED)
```

**Review by JohanMabille — CHANGES_REQUESTED** [review]
It looks like the  LibArchive vcpkg package does not provide a cmake config file

File: libmamba/CMakeLists.txt (hunk 5 above)
Comment (henryiii) [reply libmamba/CMakeLists.txt:542]: Ah, my mistake: https://cmake.org/cmake/help/latest/module/FindLibArchive.html

```suggestion
            find_package(LibArchive MODULE REQUIRED)
```

**Review by JohanMabille — APPROVED** [review]

**Comment by JohanMabille** [comment]
Thanks!

**Pull request merged**